doc/rst/technotes/gpu.rst

```diff
@@ -202,20 +223,21 @@ Chapel’s build system will automatically try and deduce what type of GPU you
 have and where your installation of relevant runtime (e.g. CUDA or ROCm) are.
 If the type of GPU is not detected you may set the ``CHPL_GPU`` environment
 variable manually to either ``nvidia`` or ``amd``.  ``CHPL_GPU`` may also
-manually be set to ``cpu`` to use `CPU as Device mode`_.
+manually be set to ``cpu`` to use `CPU-as-Device mode`_.
 
 Based on the value of ``CHPL_GPU``, Chapel's build system will also attempt to
 automatically detect the path to the relevant runtime. If it is not
 automatically detected (or you would like to use a different installation) you
 may set ``CHPL_CUDA_PATH`` and/or ``CHPL_ROCM_PATH`` explicitly.
 
-The ``CHPL_GPU_ARCH`` environment variable can be set to control the desired
-GPU architecture to compile for. The default value is ``sm_60`` for
+The ``CHPL_GPU_ARCH`` environment variable can be set to control the desired GPU
+architecture to compile for. The default value is ``sm_60`` for
 ``CHPL_GPU_CODEGEN=cuda``. You may also use the ``--gpu-arch`` compiler flag to
-set GPU architecture.  If using AMD, `this table in the LLVM documentation
-<https://llvm.org/docs/AMDGPUUsage.html#processors>`_ has possible architecture
-values (see the "processor" column). For NVIDIA, see the `CUDA Programming
-Guide
+set GPU architecture.  If using AMD, this variable must be set. `This table in
+the ROCm documentation
+<https://rocm.docs.amd.com/en/latest/release/gpu_os_support.html#linux-supported-gpus>`_
+has possible architecture values (see the "LLVM Target" column). For NVIDIA, see
+the `CUDA Programming Guide
 <https://docs.nvidia.com/cuda/cuda-c-programming-guide/#features-and-technical-specifications>`_.
```

Comment:
This isn't a new issue in your PR but for this link:

> <https://docs.nvidia.com/cuda/cuda-c-programming-guide/#features-and-technical-specifications>`_.

I guess it's better having it than not, but I'm not sure just citing it is enough for our readers to figure out what they should set `CHPL_GPU_ARCH` to. The table that shows me what compute capability is needed for certain features. But this doesn't really help me much because:

- It doesn't really tell me what compute capability my actual, physical, GPU is capable of.
- These are a bunch of CUDA features; sure maybe for a lot of these there's a 1:1 correspondence but still I want to read about things in terms of Chapel not CUDA features.
- It doesn't really tell you how to translate the compute capability to what you'd pass to `CHPL_GPU_ARCH` (I assume its just `sm_` + the number in the table without a '.', but I don't think we should expect our reader to figure that out).

More pointedly: I'm unsure why I'd ever want to use anything other than the default sm. Will I get better performance by choosing a higher sm or is the best strategy to choose the lowest sm that has whatever features I require (so the generated binary can target the widest range of GPUs)?

Anyway, I'm not sure we have answers to these, and answering them is probably more than we can reasonably take on before our doc freeze so I'd probably leave this "as is" and fork a GH issue to followup.

Reply:
Fair enough. I have been using https://developer.nvidia.com/cuda-gpus which is incrementally better. I don't know whether there's a perfect solution here.